api/v1_coin.go

```diff
@@ -13,73 +13,105 @@ func (app *ApiServer) v1Coin(c *fiber.Ctx) error {
 		})
 	}
 
+	// See v1_coins for the query explanation.
 	sql := `
-		WITH member_counts_24h_ago AS (
-			SELECT 
-				COUNT(DISTINCT balances_24h_ago.user_id) AS members,
-				balances_24h_ago.mint
-			FROM (
-				SELECT DISTINCT ON (
-						sol_token_account_balance_changes.mint, 
-						sol_token_account_balance_changes.account,
-						users.user_id, associated_wallets.user_id
-					)
-					sol_token_account_balance_changes.mint,
-					sol_token_account_balance_changes.account,
-					sol_token_account_balance_changes.balance,
-					COALESCE(associated_wallets.user_id, users.user_id) AS user_id
-				FROM sol_token_account_balance_changes
-				LEFT JOIN associated_wallets
-					ON associated_wallets.wallet = sol_token_account_balance_changes.owner
-				LEFT JOIN sol_claimable_accounts 
-					ON sol_claimable_accounts.account = sol_token_account_balance_changes.account
-				LEFT JOIN users ON users.wallet = sol_claimable_accounts.ethereum_address
-				WHERE block_timestamp < NOW() - INTERVAL '24 hours'
-					AND (associated_wallets.user_id IS NOT NULL OR users.user_id IS NOT NULL)
-					AND sol_token_account_balance_changes.mint = @mint
-				ORDER BY
-					sol_token_account_balance_changes.mint, 
-					sol_token_account_balance_changes.account,
-					users.user_id, associated_wallets.user_id, 
-					block_timestamp DESC
-			) AS balances_24h_ago
-			WHERE balance > 0
-			GROUP BY balances_24h_ago.mint
-		), member_counts AS (
+		WITH t_user_balance_changes AS (
+			SELECT
```

Comment:
I wonder if two LEFT JOINs would be faster than a union like this.

Reply:
that said I haven't done a direct comparison